backend/src/main/java/asegroup1/api/services/landregistry/LandRegistryServiceImpl.java

```diff
@@ -279,4 +289,56 @@ private Colour getColoursForNormalisedValues(Double normalisedValue) {
         //The higher the normalised value the darker the red will appear
         return new Colour(255 - (int) (normalisedValue * 200));
     }
+
+	private HashMap<String, Long> getAllPostcodePrices(String... postcodes) throws IOException, UnirestException {
+		List<LandRegistryData> transactions = getTransactions(LandRegistryQuery.buildQueryAveragePricePostcode(postcodes));
+		HashMap<String, Long> postcodePrices = new HashMap<>();
+		for (LandRegistryData data : transactions) {
+			String postcode = data.getConstraint(Selectable.postcode);
+			String priceStr = data.getConstraint(Selectable.pricePaid);
+
+			if (postcode != null && priceStr != null) {
+				try {
+					Long pricePaid = Long.parseLong(priceStr);
+					postcodePrices.put(postcode, pricePaid);
+				} catch (NumberFormatException e) {
+					e.printStackTrace();
+				}
+			}
+		}
+
+		// Map all postcodes without a average price to null
+		List<String> unmatchedPostcodes = new ArrayList<>(Arrays.asList(postcodes));
+		unmatchedPostcodes.removeAll(postcodePrices.keySet());
+
+		for (String postcode : unmatchedPostcodes) {
+			postcodePrices.put(postcode, null);
+		}
+
+
+		return postcodePrices;
+	}
+
+	public void updatePostcodeDatabase(String postcodePrefix) throws IOException, UnirestException {
+		long startTime = System.currentTimeMillis();
+		int updatedRecords = 0;
+
+		HashMap<String, List<String>> postcodeAreas = postCodeCoordinatesDao.getAllPostcodes(postcodePrefix, true, 1);
+		double numAreas = postcodeAreas.size();
+		double numDone = 0;
+
+		for (Entry<String, List<String>> postcodeArea : postcodeAreas.entrySet()) {
+
+			System.out.printf("Updating records in \"%s\" %.3f %% done, %s remaining\n", postcodeArea.getKey(), (numDone / numAreas) * 100, 
+					Duration.ofMillis(Math.round(((System.currentTimeMillis() - startTime)/numDone)*(numDone-numAreas))));
+			List<String> postcodes = postcodeArea.getValue();
+			HashMap<String, Long> newPrices = getAllPostcodePrices(postcodes.toArray(new String[postcodes.size()]));
```

Comment:
IntelliJ wants this to be `postcodes.toArray(new String[0]))`, either change this to that or change the other one so its like this, but it should be consistent.